packages/devtools_app/lib/src/debugger/flutter/codeview.dart

```diff
@@ -22,57 +22,92 @@ import 'debugger_model.dart';
 class CodeView extends StatefulWidget {
   const CodeView({
     Key key,
-    this.script,
-    this.stack,
     this.controller,
+    this.scriptRef,
     this.onSelected,
-    this.lineNumberToBreakpoint,
   }) : super(key: key);
 
   static const rowHeight = 20.0;
   static const assumedCharacterWidth = 16.0;
 
   final DebuggerController controller;
-  final Map<int, BreakpointAndSourcePosition> lineNumberToBreakpoint;
-  final Script script;
-  final vm.Stack stack;
-  final void Function(Script script, int line) onSelected;
+  final ScriptRef scriptRef;
+
+  final void Function(ScriptRef scriptRef, int line) onSelected;
 
   @override
   _CodeViewState createState() => _CodeViewState();
 }
 
 class _CodeViewState extends State<CodeView> {
+  Script script;
   List<String> lines = [];
   Set<int> executableLines = {};
+
   LinkedScrollControllerGroup verticalController;
   ScrollController gutterController;
   ScrollController textController;
 
-  // The paused positions in the current [widget.script] from the [widget.stack].
-  List<int> pausedPositions;
+  ScriptRef get scriptRef => widget.scriptRef;
 
   @override
   void initState() {
     super.initState();
 
+    _initScriptInfo();
+
     verticalController = LinkedScrollControllerGroup();
     gutterController = verticalController.addAndGet();
     textController = verticalController.addAndGet();
-    _updateLines();
-    _updatePausedPositions();
+
+    widget.controller.scriptLocation.addListener(_handleScriptLocationChanged);
+  }
+
+  void _initScriptInfo() {
+    script = widget.controller.getScriptCached(scriptRef);
+
+    if (script == null) {
+      if (scriptRef != null) {
+        widget.controller.getScript(scriptRef).then((script) {
+          setState(() {
```

Comment:
make sure the scriptRef that was specified is still the relevant scriptRef for the codeview after the async operation completes and that the widget has not been disposed.

Reply:
will do